tests/foreman/cli/test_computeresource_osp.py

```diff
@@ -0,0 +1,274 @@
+    def test_positive_osp_info(self):
+        """List the info of Openstack compute resource
+
+        :id: 8ed9f9e6-053a-402d-8f3d-3ba46300098d
+
+        :expectedresults: OSP Compute resource Info is displayed
+
+        :CaseImportance: Critical
+        """
+        name = gen_string('utf8')
+        compute_resource = make_compute_resource({
+            u'name': name,
+            u'provider': 'Openstack',
+            u'user': self.username,
+            u'password': self.password,
+            u'tenant': self.tenant,
+            u'url': self.current_osp_url
+        })
+        self.assertEquals(compute_resource['name'], name)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_delete_by_name(self):
+        """Delete the Openstack compute resource by name
+
+        :id: 8c581100-4606-4d21-a286-930fb3a7ecd8
+
+        :expectedresults: Compute resource is deleted
+
+        :CaseImportance: Critical
+        """
+        comp_res = make_compute_resource({
+            u'provider': 'Openstack',
+            u'user': self.username,
+            u'password': self.password,
+            u'tenant': self.tenant,
+            u'url': self.current_osp_url
+        })
+        self.assertTrue(comp_res['name'])
+        ComputeResource.delete({'name': comp_res['name']})
+        result = ComputeResource.exists(search=('name', comp_res['name']))
+        self.assertFalse(result)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_delete_by_id(self):
+        """Delete the Openstack compute resource by id
+
+        :id: f464429f-a4ac-4504-b009-5f56f9d29317
+
+        :expectedresults: Compute resource is deleted
+
+        :CaseImportance: Critical
+        """
+        comp_res = make_compute_resource({
+            u'provider': 'Openstack',
+            u'user': self.username,
+            u'password': self.password,
+            u'tenant': self.tenant,
+            u'url': self.current_osp_url
+        })
+        self.assertTrue(comp_res['name'])
+        ComputeResource.delete({'id': comp_res['id']})
+        result = ComputeResource.exists(search=('name', comp_res['name']))
+        self.assertFalse(result)
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_create_osp_with_url(self):
+        """Openstack compute resource negative create with invalid values
+
+        :id: a6be8233-2641-4c87-8563-f48d6efbb6ac
+
+        :expectedresults: Compute resource is not created
+
+        :CaseImportance: Critical
+        """
+        with self.assertRaises(CLIReturnCodeError):
+            ComputeResource.create({
+                u'provider': 'Openstack',
+                u'user': self.username,
+                u'password': self.password,
+                u'tenant': self.tenant,
+                u'url': 'invalid url'
+            })
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_create_with_same_name(self):
+        """Openstack compute resource negative create with the same name
```

Comment:
Attempt to create Openstack compute resource with the same name as an existing one